v2/options.go

```diff
@@ -238,3 +239,12 @@ func AllowDuplicatePuts(allow bool) Option {
 		o.BlockstoreAllowDuplicatePuts = allow
 	}
 }
+
+// WithAbsoluteIndexOffsets is a read option which interprets index offsets as
+// absolute offsets from the beginning of the CARv2 file, instead of relative
+// offsets from the beginning of the Carv1 header.
+func WithAbsoluteIndexOffsets(absolute bool) Option {
```

Comment:
This would violate the [specification of CARv2](https://ipld.io/specs/transport/car/carv2/#index-format). Is there any chance we can capture this complexity inside boost? I am sympathetic that this might be hard to do in the existing Boost implementation, but am worried that this would open a can of worms for all sorts of CARv2 parsing errors since the specification dictates offset in index format is always from the beginning of inner CARv1. 

If not, would it make sense to add this as a characteristic in CARv2 header and spec it out that way?